### DISCUSSION TITLE
By using 'oidc' connector, who will be issuing the id token?

### DISCUSSION THREAD

**Discussion by hopeshared-lee** [dexidp/dex#3289] [Q&A] [answered]
Hi, I'm using oidc connector to connect with another server (ServerA) for authentication.

It looks like the 'login' will be handled by ServerA but the ID token is issued and signed by DEX. Is that true?
Then what data from ServerA will be used to generate the ID token?

Thanks!

**Comment by nabokihms** [comment, marked as answer]
That's true. Dex takes claims from the upstream provider token and issues its token based on these claims. See the `claimMapping` section here https://dexidp.io/docs/connectors/oidc/
